tech-specs/src/explore/design/dkg.md

```diff
@@ -1 +1,154 @@
+produce the next public encryption key. The first block determines the 
+partition of weight shares of the validators and the last aggregates the 
+PVSS instances from the dealers leading to the final key.
+
+In practice, it may take even longer as the protocol waits for enough PVSS 
+instances to get onto the blockchain. Depending on the chosen refresh rate
+of the keys, another instance of the DKG protocol may need to be started 
+before the current one finishes. This is possible so long as all data
+posted for a DKG instance includes the session id \\( \tau \\).
+
+This means that the storage of a node must be adapted. Each instance of a
+DKG protocol from Ferveo is a state machine that keeps all necessary data
+from the protocol instance. We thus need to store the following data
+ - A list of sesssion ids from the last completed DKG instance up until the
+   latest started (we may prune ids for failed instances however) 
+ - The DKG state machine associated to each of these session ids.
+
+##  The DKG state machine
+
+We first describe each new message type needed for the protocol.
+- `StartDKG`: A message from the current block proposer to start the protocol
+- `AnnounceStake`: Validators participating in this instance of the DKG
+  broadcast their stake
+- `PartitionShares`: Once the block containing all of the `AnnounceStake`
+  messages are finalized, the set of participants is fixed and the
+  partition of the weight shares is added to the block.
+- `Deal`: This message contains the PVSS transcript from a validator
+  acting as a dealer
+- `Aggregate`: Combine the PVSS instances into a single one, thereby
+  producing the final key.
+
+In Ferveo, the DKG protocol is modelled as a state machine. This allows a
+validator to initialize an instance and drive it forward by reacting to
+each of the above message in the appropriate way. All necessary state
+is persisted by this state machine. We describe the  appropriate action for
+each of the above transactions.
+
+ - `StartDKG`: Initialize a fresh instace of the DKG protocol from Ferveo.
+   Then send out an `AnnounceStake` transaction.
+ - `AnnounceStake`: These simply need to be verified. The current block
+   proposer adds a `PartitionShares` message to the block after the block starting
+   the DKG instance. It is based off all the `AnnounceStake` transactions
+   which will also be included in the next block (for verification).
+ - `PartitionShares`: Validators verify the partitions and weights and
+   should extract their weight shares and store them in the DKG state machine.
+   Then they should send out `Deal` transactions.
+ - `Deal`: These need to be verified. The current block proposer checks if 
+   enough PVSS instances have been reached to meet the security threshold.
+   If so, it adds a `Aggregate` transaction to the block. Otherwise, the
+   next block proposer will take on the job. 
```

Comment:
If I understand correctly, this bit suggests that **only** an `Aggregate` transaction will be posted to the blockchain. 
That seems contradicting to (or I'm missing something)  https://github.com/anoma/anoma/pull/360/files#diff-0154dfbaff793ae84655cf88ba919b9136d89a979d9065a50e95456516a7283dR49

Reply:
The PVSS instances must be posted to the blockchain; otherwise there is no guarantee that a validator will receive enough information to properly verify the "Aggregate" transaction